### DISCUSSION TITLE
GO binaries should use the detection logic used by Govulncheck

### DISCUSSION THREAD

**Discussion by tamalsaha** [aquasecurity/trivy#5625] [Ideas]
### IDs

CVE-2023-2253

### Description

Currently trivy will detect a lot of vulnerabilities that are not present in the binary according to [Govulncheck](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck). This false +ve reports are causing confusion among users. Is it possible to trivy to use Govulncheck style alg to detect only applicable CVEs in case of Go binaries?

### Reproduction Steps

```bash
1.
2.
3.
...
```

### Target

Container Image

### Scanner

Vulnerability

### Debug Output

```bash
N/A
```

### Version

```bash
N/A
```

### Checklist

- [X] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [X] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by nikpivkin** [comment]
Hi @tamalsaha !

There is already an issue with the `govulncheck` integration request https://github.com/aquasecurity/trivy/issues/2845
